src/Pinch/Internal/FoldList.hs

```diff
@@ -67,10 +66,6 @@ fromFoldable :: F.Foldable f => f a -> FoldList a
 fromFoldable l = FoldList (\k r -> F.foldl' k r l)
 {-# INLINE fromFoldable #-}
 
-null :: FoldList a -> Bool
-null (FoldList f) = f (\_ _ -> False) True
```

Comment:
There's no reason this couldn't be implemented using foldr. I think it didn't matter at the time. Created https://github.com/abhinav/pinch/issues/5 to eventually switch.